201/ReadMe.md

```diff
@@ -1,92 +1,185 @@
 
-This sample shows you how to use utilize Azure Key Vault, Azure Container Registry and Azure Arc enabled Key Vault Extension to dynamically pull ACR credentials from AKV and pull image from ACR.
 ## Prerequisites
-* Compatible cluster, aks and aks ee is recommended 
-* <a href = "https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/reference">AZD</a>
+
+### Required
+- A native K8S cluster connected to Arc. The Azure Arc VS code extension can help to create an AKS EE cluster and connect to Arc if needed.
+- [AZS CLI](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/reference)
+
+
+### Recommended
+- VS code with [Azure Arc Extension](https://marketplace.visualstudio.com/search?term=azure%20arc&target=VSCode&category=All%20categories&sortBy=Relevance)
+- An AKS or AKS EE cluster that's connected to Arc to minimize compatibility issues
+  > The Azure Arc Extension can help creating AKS EE clusters and connecting compatible K8S clusters to Arc.
 
 ## Steps
+> Note: Arc-enabled KV steps provided here are summerized from the [full and most up-to-date instructions](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-akv-secrets-provider). Check it out if you'd like to learn more.
 
-* (Optional - Skip if already has an SP) Follow the steps to <a href="https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal">create a service principal</a> in Azure. Take note of the Client ID, tenantID and Client Secret generated in this step.
+1. A service principal with password secret is required.
+    - If you don't have one yet, follow [these steps](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal) to create an SP with password (not with certificate).
+    - If you'd like to reuse an existing SP, make sure it has a password secret. You can associate a new secret with your existing SP with these [instructions](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-3-create-a-new-application-secret).
+  
+    Please take note of the Client ID (a.k.a., App ID), tenantID, service principal object ID and Client Secret generated in this step.
 
-* Use the client ID and Client Secret from the first step to create a Kubernetes secret on the connected cluster:
-```
-kubectl create secret generic secrets-store-creds --from-literal clientid="<client-id>" --from-literal clientsecret="<client-secret>"
+2. Use the client ID and client secret as plain text from the first step to create a Kubernetes secret on the connected cluster:
+    ```
+    kubectl create secret generic secrets-store-creds --from-literal clientid="<client-id>" --from-literal clientsecret="<client-secret>"
```

Comment:
Looks like we are not feeding the secret name into helm. Please stress that the secret name should not be modified.

Reply:
Added a comment to step 9